libs/event/src/lib/pages/analytics/analytics.component.ts

```diff
@@ -48,30 +58,28 @@ export class AnalyticsComponent implements OnInit {
       switchMap((eventId: string) => this.service.valueChanges(eventId)),
       tap(async event => {
         this.eventType = event.type;
-        const analytics = await this.service.queryAnalytics(event.id);
 
-        // transform the analytics records
-        this.analytics = analytics.eventUsers.map(analytic => {
-          const transformedAnalytic = {
-            ...analytic,
-            name: `${analytic.firstName} ${analytic.lastName}`,
-          };
 
-          // add watch time to the analytic record
-          if (this.eventType === 'screening') {
-            // retrieve watch time from invitation
-            const [invitation] = this.invitationQuery.getAll({
-              // we are looking for invitation between this event and this user id
-              filterBy: invit => invit.eventId === analytic.eventId &&
-                (
-                  invit.toUser?.uid === analytic.userId ||
-                  invit.fromUser?.uid === analytic.userId
-                )
-            });
-            transformedAnalytic.watchTime = invitation?.watchTime ?? 0;
-          }
-
-          return transformedAnalytic;
+        const invitations = this.invitationQuery.getAll({
+          // we are looking for invitations related to this event
+          filterBy: invit => invit.eventId === event.id && !!invit.watchTime
+        });
+
+        const orgIds = Array.from(new Set(invitations.map(i => i.fromUser?.orgId || i.toUser?.orgId).filter(orgId => !!orgId)));
```

Comment:
Create intermediary variables :
```suggestion
        const allOrgIds = invitations.map(i => i.fromUser?.orgId || i.toUser?.orgId).filter(orgId => !!orgId);
        const orgIds = Array.from(new Set());
```